### PR TITLE
api: seperate internal client into internalapi

### PR DESCRIPTION
**Pull request by bobheadxi**
Extracted from and required by #28117. Users of just the API types no longer need to import `httpcli` and a whole bunch of other things, which was causing some awkward import cycles (see #28117)

### CODE REVIEW

File: internal/repos/phabricator.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/inconshreveable/log15"
 
 	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/api/internal_api"
```
Comment (eseliger) [inline internal/repos/phabricator.go:13]: I think I personally would prefer 
```suggestion
	internalapi "github.com/sourcegraph/sourcegraph/internal/api/internal"
```

but no strong preference

Comment (bobheadxi) [reply internal/repos/phabricator.go:13]: I think naming it `internal` will prevent packages outside of `internal/api` from importing this, no?

Comment (eseliger) [reply internal/repos/phabricator.go:13]: oh lord 🙈 yeah probably. I think I am mainly concerned about the `_` not so much the repetitiveness in the path :)

Comment (bobheadxi) [reply internal/repos/phabricator.go:13]: OH hey it works!

Comment (bobheadxi) [reply internal/repos/phabricator.go:13]: Oh nevermind it doesn't:

```
enterprise/internal/codemonitors/email/email.go:12:2: use of internal package github.com/sourcegraph/sourcegraph/internal/api/internal not allowed (typecheck)
        internalapi "github.com/sourcegraph/sourcegraph/internal/api/internal"
```

Comment (bobheadxi) [reply internal/repos/phabricator.go]: I couldn't find another package in the repo with an underscore name, so I renamed to `/internalapi` for consistency 5780bbed5a2bbb686ee795983acf7136039f3d68

File: cmd/gitserver/server/gitolite_phabricator_test.go [hunk 2]
```diff
@@ -6,7 +6,7 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/api/internalapi"
```
Comment (ryanslade) [inline cmd/gitserver/server/gitolite_phabricator_test.go:9]: nit: Would it work if it was instead `sourcegaph/internal/internalapi`? 

Having `internalapi` under `api` feels like too many levels

**Review by ryanslade — APPROVED** [review]

File: cmd/gitserver/server/gitolite_phabricator_test.go (hunk 2 above)
Comment (bobheadxi) [reply cmd/gitserver/server/gitolite_phabricator_test.go:9]: I think it makes sense still to have it under `api`, as `internalapi` is highly dependent on `api`

**Pull request merged**